tianshou/data/batch.py

```diff
@@ -36,8 +37,11 @@ def _is_scalar(value: Any) -> bool:
     # 3. python object rather than dict / Batch / tensor
     # the check of dict / Batch is omitted because this only checks a value.
     # a dict / Batch will eventually check their values
-    value = np.asanyarray(value)
-    return value.size == 1 and not value.shape
+    if isinstance(value, (np.ndarray, torch.Tensor)):
+        return value.size == 1 and not value.shape
```

Comment:
for torch.Tensor, use ``value.numel()``
```python
import torch
a = torch.ones((3, 4))
a.size
Out[15]: <function Tensor.size>
a.numel()
Out[16]: 12
```

Reply:
Indeed. I thought it was the same for both. FIxed